test/unit/test_logger.py

```diff
@@ -126,3 +126,28 @@ def test_cyan_text():
     x = '{}{}{}'.format(colorama.Fore.CYAN, 'foo', colorama.Style.RESET_ALL)
 
     assert x == logger.cyan_text('foo')
+
+
+def test_markup_detection_pycolors_0(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '0')
+    assert not logger.should_do_markup()
+
+
+def test_markup_detection_pycolors_1(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '1')
+    assert logger.should_do_markup()
+
+
+def test_markup_detection_is_tty(monkeypatch, mocker):
+    monkeypatch.delenv('PY_COLORS', raising=False)
+    monkeypatch.setenv('TERM', 'xterm')
+    stdout_mock = mocker.patch("sys.stdout")
+    stdout_mock.isatty.return_value = True
+    assert logger.should_do_markup(sys.stdout)
```

Comment:
```suggestion
    with monkeypatch.ctx() as mp_ctx:
        mp_ctx.setattr('sys.stdout.isatty', lambda: True)
        assert logger.should_do_markup(sys.stdout)
```

Reply:
@ssbarnea this is the correct usage of context. just apply it and it should be good to go like this.